src/app/map/map.component.ts

```diff
@@ -794,9 +823,59 @@ export class MapComponent implements OnInit {
         this.map.on('zoomstart', () => {
             this.previousZoom = this.map.getZoom();
         });
+
+        this.streamgageService.streamGageMarkers.on('click', (e) => {
+            this.queryStreamGageGraph(e);
+        })
+
+        //Get the value of the current zoom
+        this.map.on('zoomend dragend', () => {
+            this.currentZoom = this.map.getZoom();
+
+            //Disable checkbox on zoom < 9
+            if(this.currentZoom < 9){
+                document.querySelectorAll('.leaflet-control-layers.leaflet-control').forEach(control => {control.addEventListener('mouseover', this.disableStreamGage)});
+            }else{
+                document.querySelectorAll('.leaflet-control-layers.leaflet-control').forEach(control => {control.removeEventListener('mouseover', this.disableStreamGage)});
+            }
+
+            //Hide stream gages if layer is checked and zoom < 9
+            //Need to do this because minzoom cannot be set on L.FeatureGroup
+            if (this.streamgagesVisible && this.currentZoom < 9){
+                if (this.map.hasLayer(this.streamgageService.streamGageMarkers)){
+                    this.streamgageService.streamGageMarkers.clearLayers();
+                    // if zooming too fast from zoom > 9 to 8, need to wait for gages to finish loading before clearing layers
+                    setTimeout(() => {
+                        if (this.streamgagesVisible && this.currentZoom < 9){
+                            this.streamgageService.streamGageMarkers.clearLayers();
+                        }
+                    }, 1500)
+                    setTimeout(() => {
+                        if (this.streamgagesVisible && this.currentZoom < 9){
+                            this.streamgageService.streamGageMarkers.clearLayers();
+                        }
+                    }, 3000)
+                    setTimeout(() => {
+                        if (this.streamgagesVisible && this.currentZoom < 9){
+                            this.streamgageService.streamGageMarkers.clearLayers();
+                        }
+                    }, 7000)

```

Comment:
I'd like to avoid these timeouts if possible. Computer speeds, browsers, and networks all play a role in how long it takes elements/calls/etc to load on a user's computer. So it's dangerous to try to accomplish things with timeouts. That said they are sometimes unavoidable (as you'll notice elsewhere in this codebase). Let me know if you'd like me to explore options here. Without digging into it I'm wondering if there is a listener we can use to know when the gages have finished loaded and it's safe to clear.